share/spice/timer/timer.js

```diff
@@ -7,290 +7,430 @@ License: CC BY-NC 3.0 http://creativecommons.org/licenses/by-nc/3.0/
 (function (env) {
     'use strict';
 
-    var started = false;
-    var MAX_TIME = 59999; // => 999m59s
+    var MAX_TIME = 359999, // => 99 hrs 59 mins 59 secs
+        soundUrl = DDG.get_asset_path('timer', 'alarm.mp3'),
+        Timer;
 
-    env.ddg_spice_timer = function(api_result) {
-        
-        //prevent flash of unstyled content if spice shows before css loads
-        function waitForCss() {
-            var interval = setInterval(findTimerCss, 10),
-                tries = 0;
-
-            function findTimerCss() {
-                var el = $('link[href$="timer.css"]')[0];
-                tries++;
-              
-                if (el) {
-                    clearInterval(interval);
-                    if (el.sheet) {
-                        //already loaded, no need for 'load' event
-                        showTimer();
-                    } else {
-                        //not yet loaded, attach a 'load' event
-                        $(el).load(showTimer);
-                    }
+    // helper methods
+
+    // add zeros to beginning of string
+    function padZeros(n, len) {
+        var s = n.toString();
+        while (s.length < len) {
+            s = '0' + s;
+        }
+        return s;
+    }
+
+    // split seconds into hours, minutes, seconds
+    function getHrsMinsSecs(totalSeconds) {
```

Comment:
Possibly, but using a library might add extra overhead and we're running this every 100ms. Also, Moment.js hadn't been added to the libraries when I worked on this last :)